src/index.js

```diff
@@ -5,20 +5,37 @@ function notifyAirbrake(airbrake, options = {}) {
   return function(ctx, next) {
     return next().catch((err)=> {
       const req = ctx.request;
-      const res = ctx.response;
+      
+      const notice = {};
 
-      err.url = req.url;
-      err.action = req.url;
-      err.component = err.component || defaultComponent;
-      err.httpMethod = req.method;
-      err.params = req.body;
-      err.session = req.session;
-      err.ua = req.headers['User-Agent'];
+      notice.error = err;
 
-      airbrake.notify(err);
+      // Some libraries like Axios directly annotate error object 
+      // with contextual info such as error code, req url. 
+      // Add these to `environment` field of the notice.
+      // These go in the `environment` and not in the `context` field
+      // because that's how we used to send notice to old Airbrake.
+      const keys = Object.keys(err);
+      notice.environment = {};
+      keys.forEach((key)=> {
+        notice.environment[key] = err[key];
+      });
```

Comment:
I'm a little bit nervous about just copying all of the Error's keys.  Perhaps this should be handled via an airbrake filter instead?  Ex:

```javascript
const airbrake = new Client(...);

airbrake.addFilter(function(notice) {
  const err = notice.error;

  // Shallow-copy the Error's fields into notice.environment
  notice.environment = Object.assign({}, err);

  return notice;
});

const mw = notifyAirbrake(airbrake);
```

Would this accomplish the same thing?

Reply:
A filter serves a different purpose no? It is to transform existing notices?

Reply:
Right.  But, this logic really seems like an application-specific concern to me.  Additionally, this also introduces a security concern since 3rd-party libraries might embed sensitive information in their Error objects.  Ex: a 3rd-party `AuthenticationError` might include sensitive information such as API keys and session identifiers.  If the developer was unaware of these fields (ex: they were undocumented), then Airbrake would begin collecting this sensitive information.